File: ProblemSet/111. 爬楼梯/python/111. 爬楼梯 公式法.py
class Solution:
    """
    @param n: An integer
    @return: An integer
    """
    def climbStairs(self, n):
        if n == 0:
            return 0
        gloden_ratio = (1 + 5 ** 0.5) / 2
        return round(gloden_ratio ** (n + 1) / (5 ** 0.5))

File: ProblemSet/111. 爬楼梯/java/111. 爬楼梯 公式法.java
public class Solution {
    /**
     * @param n: An integer
     * @return: An integer
     */
    public int climbStairs(int n) {
        if (n == 0) return 0;
        double glodenRatio = (1 + Math.sqrt(5)) / 2;
        return (int)Math.round(Math.pow(glodenRatio, n + 1) / Math.sqrt(5));
    }
}